onnxruntime/test/mlas/bench/bench_rope.cpp

```diff
@@ -4,42 +4,48 @@
 #include "mlas.h"
 #include "benchmark/benchmark.h"
 #include "bench_util.h"
+#include "core/framework/float16.h"
 
+using namespace onnxruntime;
+
+template <typename T>
 void RunRoPEBenchmark(size_t rotary_emb_dim, bool interleaved, benchmark::State& state) {
   const float Pi = 2 * std::acos(0.0f);
 
-  std::vector<float> input(rotary_emb_dim);
+  std::vector<T> input(rotary_emb_dim);
   size_t table_len = interleaved ? rotary_emb_dim / 2 : rotary_emb_dim;
-  std::vector<float> sin_data(table_len);
-  std::vector<float> cos_data(table_len);
-  std::vector<float> output_ref(rotary_emb_dim), output_impl(rotary_emb_dim);
+  std::vector<T> sin_data(table_len);
+  std::vector<T> cos_data(table_len);
+  std::vector<T> output_ref(rotary_emb_dim), output_impl(rotary_emb_dim);
 
   for (size_t i = 0; i < rotary_emb_dim; ++i) {
-    input[i] = static_cast<float>(i + 1);
+    input[i] = static_cast<T>(1.f * i + 1);
   }
   for (size_t i = 0; i < table_len; ++i) {
     float theta = (float)i / 1000 * Pi;
```

Comment:
1. Could you use static_cast<float> instead?
2. Should the code be changed to match the code in test_rope.cpp?